Volume/Views/MainView/MagazinesList.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  MagazinesList.swift
+//  Volume
+//
+//  Created by Hanzheng Li on 3/4/22.
+//  Copyright © 2022 Cornell AppDev. All rights reserved.
+//
+
+import SwiftUI
+
+struct MagazinesList: View {
+    @State private var state: MainView.TabState<Results> = .loading
+    
+    private func fetchContent(_ done: @escaping () -> Void = { }) {
+        // simulate network request delay
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            // TODO: add actual network request
+            done()
+            state = .results((
```

Comment:
Why don't you need self here?

Reply:
not sure, but if I had to guess, it might have something to do w/ SwiftUI views being value types instead of reference types